GliaWidgets/Sources/CallVisualizer/ScreenSharing/ViewController/ScreenSharingViewController.swift

```diff
@@ -1,38 +1,41 @@
 import UIKit
+import SwiftUI
 
 extension CallVisualizer {
     final class ScreenSharingViewController: UIViewController {
-        private lazy var screenSharingView = ScreenSharingView(props: props.screenSharingViewProps)
-        private var props: Props
+        let model: ScreenSharingView.Model
 
         // MARK: - Initialization
 
-        init(props: Props) {
-            self.props = props
-            super.init(nibName: "", bundle: nil)
+        init(
+            model: ScreenSharingView.Model
+        ) {
+            self.model = model
+            super.init(nibName: nil, bundle: nil)
         }
 
+        @available(*, unavailable)
         required init?(coder: NSCoder) {
             fatalError("init(coder:) has not been implemented")
         }
 
         // MARK: - View lifecycle
 
-        override func loadView() {
-            view = screenSharingView
-        }
-
         override func viewDidLoad() {
             super.viewDidLoad()
-            renderProps()
+            let hostingController: UIHostingController<ScreenSharingView>
+            let screenSharingView = ScreenSharingView(model: model)
+            hostingController = UIHostingController(rootView: screenSharingView)
+            addChild(hostingController)
+            view.addSubview(hostingController.view)
+            hostingController.didMove(toParent: self)
```

Comment:
to keep consistency do we need to call `willMove` method also?

Reply:
You're right. Calling willMove(toParent:) ensures that we're following the recommended pattern for view controller containment consistently. I've added that call before adding the child view controller. Thanks for pointing it out!

Reply:
fixed [2f45a67](https://github.com/salemove/ios-sdk-widgets/pull/784/commits/2f45a67ef85d8be264e9fdf761f0925da56dfc7b)